### ISSUE TITLE
Running RStudio Server with unprivileged user not possible

### ISSUE THREAD

**Issue by antonneubauer** [rocker-org/rocker-versioned2#592]
We ran into a problem trying to start the rocker/rstudio image with [enroot](https://www.pugetsystems.com/labs/hpc/Run-Docker-Containers-with-NVIDIA-Enroot-2142/) on our slurm-cluster. Since our users are unprivileged, we get the following error:
`2023-01-11T19:23:56.907205Z [rserver] ERROR Attempt to run server as user 'rstudio-server' (uid 999) from account '<userAccount>' (uid 11293) without privilege, which is required to run as a different uid; LOGGED FROM: virtual rstudio::core::ProgramStatus rstudio::server::Options::read(int, char* const*, std::ostream&) src/cpp/server/ServerOptions.cpp:318
/usr/lib/rstudio-server/bin/rstudio-server: line 7: systemctl: command not found`

We have fixed this by copying your Dockerfile, inserting `RUN echo "server-user=$USER" >> /etc/rstudio/rserver.conf` before `CMD ["/init"]` and building our own image.

We were wondering, if this is something you might want to/ can to add to your Dockerfiles.
If there's something that speaks against doing that I would be very interested in learning about it since I am fairly new to this field.

**Comment by eitsupi** [comment]
I believe that the RStudio Server started by the `/init` command is intended to be run by the root user, so it does not work correctly by a non-root user.

> We have fixed this by copying your Dockerfile, inserting `RUN echo "server-user=$USER" >> /etc/rstudio/rserver.conf` before `CMD ["/init"]` and building our own image.

I don't think you have to do it. A simple Dockerfile like the following should should be sufficient:

```dockerfile
FROM rocker/rstudio:4
RUN echo "server-user=$USER" >> /etc/rstudio/rserver.conf
```

**Comment by antonneubauer** [comment]
Thanks for the quick reply!
Looks like your (much simpler) version of the Dockerfile works. Thank you for that.

We are currently testing the set up on our Cluster and I'll report back if we run into issues because of the non-root-start up.

**Comment by eitsupi** [comment]
I'm glad to hear that.
I'm closing this for now, but feel free to post again if you have any more problems.

**Comment by lambdamoses** [comment]
Hi, I got the same problem when running RStudio from the rocker container on a server using apptainer. I translated your short dockerfile into apptainer definition (though had to hard code my user name since somehow `$USER` is blank when building the image with fakeroot) and it worked, but that's one extra annoying step. Is there a reason not to add `RUN echo "server-user=$USER" >> /etc/rstudio/rserver.conf` in the rocker docker file? Anyway, I think I'm going to make my own docker image to run my R package in RStudio inside a container and I'll add that line, unless you advise not to.
